### DISCUSSION TITLE
Set open to false after a delay on useClick

### DISCUSSION THREAD

**Discussion by joceqo** [floating-ui/floating-ui#2194] [React]
Hi, first of all thank you for making this package it's really helpful 🙏. 
I can't find a way using it to use the click interaction in order to have something pop up and close itself after an amount of time, i guess useCustomLogic could be a solution

**Comment by atomiks** [comment]
You can just use an effect when the open state becomes `true` and use a timeout to set the state to false.

**Reply by joceqo** [reply]
I did that(I should have precise,sorry) but I struggle more to nicely handle the clearTimeout if the user click again. Because I have no way / event to tell me if the user already clicked aside isOpen

**Comment by joceqo** [comment]
i think i found out, using toggle on the click with the setTimeout
